elide-async/src/main/java/com/yahoo/elide/async/service/DefaultResultStorageEngine.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.async.service;
+
+import com.yahoo.elide.ElideSettings;
+import com.yahoo.elide.async.models.AsyncQuery;
+import com.yahoo.elide.async.models.AsyncQueryResultStorage;
+import com.yahoo.elide.core.DataStore;
+import com.yahoo.elide.request.EntityProjection;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.sql.Blob;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.Iterator;
+
+import javax.inject.Singleton;
+import javax.sql.rowset.serial.SerialBlob;
+
+/**
+ * Default implementation of ResultStorageEngine.
+ * It uses DB to store results.
+ */
+@Singleton
+@Slf4j
+@Getter
+public class DefaultResultStorageEngine implements ResultStorageEngine {
+    private static final String FORWARD_SLASH = "/";
+
+    @Setter private ElideSettings elideSettings;
+    @Setter private DataStore dataStore;
+
+    public DefaultResultStorageEngine() {
+    }
+
+    public DefaultResultStorageEngine(ElideSettings elideSettings, DataStore dataStore) {
+        this.elideSettings = elideSettings;
+        this.dataStore = dataStore;
+    }
+
+    @Override
+    public URL storeResults(String asyncQueryID, byte[] byteResponse, String downloadBaseURL) {
+
+        try {
+            Blob response = new SerialBlob(byteResponse);
+
+            DBUtil.executeInTransaction(elideSettings,
+                    dataStore, (tx, scope) -> {
```

Comment:
This may not be safe because we are asking an Elide transaction to manipulate a model that is not in the dictionary.